### ISSUE TITLE
Regression: Cromwell does not restart externally killed jobs on the SFS backend.

### ISSUE THREAD

**Issue by rhpvorderman** [broadinstitute/cromwell#4998]
Can be reproduced using the following workflow

```wdl
version 1.0

task crash {
  command <<<
    kill -9 $$
  >>>
}

workflow crash {
  call crash 
}

```
We use a configuration with the following values:
```HOCON
backend {
  default="SGE"
  providers {
    SGE {
      actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"
      config {
        exit-code-timeout-seconds = 120
        default-runtime-attributes {
          maxRetries: 2
        }
      }
    }
  }
}
workflow-options {
  workflow-failure-mode = "ContinueWhilePossible"
}
```
On Cromwell 37 the workflow will be run. Jobs will be killed and retried.
On Cromwell 39, the retries will not happen any more.
This is very annoying, as our cluster kills jobs that exceed the memory limit, and some java based jobs seem to have random memory spikes. Having only 1 try means basically that a workflow with 50-100 jobs will usually fail, unless we give some jobs an insane memory parameter.

This is probably caused by the refactoring in:
https://github.com/broadinstitute/cromwell/pull/4654/files
EDIT: This statement was not meant to put a blame on someone. I understand that code needs to be refactored at times and that bugs can creep in.

I will look if I can fix the issue myself but maybe @cjllanwarne can also have a quick look? That would be much appreciated!

**Comment by rhpvorderman** [comment]
I did some more testing. The same workflow, but now with the command `exit 1` is properly retried by cromwell 39. 
Also, the code in https://github.com/broadinstitute/cromwell/pull/4654/files looks sound, it must be elsewhere. Somewhere in the code it decides that "externally killed" jobs should not be retried but other failed jobs can be retried.

**Comment by rhpvorderman** [comment]
Found it https://github.com/broadinstitute/cromwell/blob/937cb05b6d6b438d5726f479beda2d698e3a269a/supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemAsyncJobExecutionActor.scala#L218

Not having an exit code while the job is stopped is marked as nonretryable. I can make a PR changing this to retryable. If that is the proper way of solving this. What are your thoughts @cjllanwarne ?

**Comment by cjllanwarne** [comment]
Hey @rhpvorderman - given your experiences, it sounds like this would be a valid change. A PR would be a great idea if you can see what to do and can test that it's working. Thanks!

**Comment by rhpvorderman** [comment]
@cjllanwarne I pointed to the wrong line of code, sorry for that. I have identified the bug.

The refactoring produced **better** code. The code written before the refactoring created a rc file with exit code `9` instead of `137` (SIGKILL). If cromwell gets SIGINT (`130`) , SIGKILL (`137`) or SIGTERM(`143`) as exit codes for a job, it assumes that cromwell was the one that aborted them and the jobs should NOT be retried. This makes perfect sense. 

The refactored code now returns a return code(143) that makes cromwell believe that the job should not be retried. My solution would be to write a non-sensical return code in the case exit-timeout-seconds is used. I am working on a pr now.
